lib/Echidna/Mutator.hs

```diff
@@ -0,0 +1,86 @@
+module Echidna.Mutator where
+
+import Control.Monad.Random.Strict (fromList, MonadRandom, getRandomR)
+import Data.Maybe (maybe)
+import Data.ByteString (ByteString, pack, unpack)
+import GHC.Word (Word8)
+import EVM.ABI (AbiValue(..))
+
+listMutators :: MonadRandom m => m ([a] -> m [a])
+listMutators = fromList [(return, 1), (expandRandList, 10), (deleteRandList, 10), (swapRandList, 10)] 
+ 
+mutateV :: MonadRandom m => Maybe Int -> [AbiValue] -> [AbiValue] -> m [AbiValue]
+mutateV mn fs vs = do
+  f <- listMutators
+  xs <- f vs
+  return $ maybe xs (`take` (xs ++ fs)) mn
+
+mutateBS :: MonadRandom m => Maybe Int -> [Word8] -> ByteString -> m ByteString
+mutateBS mn fs bs = do
+  f <- listMutators
+  xs <- f $ unpack bs
+  return $ pack $ maybe xs (`take` (xs ++ fs)) mn
+
+
+replaceAt :: a -> [a] -> Int -> [a]
```

Comment:
you can use lens to do that `(element index .~ value) list`

Reply:
> Might be good to actually use vector, there is a lot of random access going on.

The use of list makes easier to write mutation functions to do interesting things like expand or splice.

Regarding the use of lens, I'm not confident enough to re-write this code using that library. If someone wants try it, just commit in this branch.

Reply:
under the hood all the evm abi stuff is `Vector`. we should try to use `Vector` whenever possible if those arrays are being passed directly into these functions.